src/main/scala/mesosphere/marathon/upgrade/DeploymentActor.scala

```diff
@@ -123,7 +123,7 @@ private class DeploymentActor(
   def scaleRunnable(runnableSpec: RunSpec, scaleTo: Int,
     toKill: Option[Iterable[Instance]],
     status: DeploymentStatus): Future[Unit] = {
-    val runningTasks = taskTracker.specInstancesLaunchedSync(runnableSpec.id)
+    val runningInstances = instanceTracker.specInstancesLaunchedSync(runnableSpec.id)
```

Comment:
It does not. 
As ww made the TaskTracker an Actor and used the sync form in all existing places.
We need to replace all occurrences of the sync form - but given the time constraints we have, I would make this an extra task.